olive/passes/onnx/graph_surgeries.py

```diff
@@ -447,6 +452,49 @@ def __call__(self, model: ModelProto):
         return self._add_zero_point(model, zero_point_value, zero_point_onnx_dtype, zero_point_np_dtype)
 
 
+class QuickGeluToSigmoid(Surgeon):
+    """Lower QuickGelu operator to standard ONNX operators.
+
+    QuickGelu pattern:
+    [Input] --> QuickGelu --> [Output]
+
+    Replaced with:
+    [Input] --> Mul --> Sigmoid --> Mul --> [Output]
+                 |                    ^
+                 |                    |
+                 +--------------------+
+
+    Where the first Mul multiplies by alpha (default 1.702).
+    QuickGelu(x) = x * sigmoid(alpha * x)
+    """
+
+    ALPHA = 1.702  # QuickGelu default alpha
+
+    def __init__(self):
+        super().__init__()
+        self._rule = pattern.RewriteRule(
+            self._target_pattern,
+            self._replacement_pattern,
+        )
+
+    def _target_pattern(self, op, x):
+        return op.QuickGelu(x, _domain=MSFT_DOMAIN)
+
+    def _replacement_pattern(self, op, x):
+        # Create alpha constant
+        alpha = op.Constant(value_float=self.ALPHA)
```

Comment:
You may also want to cast alpha to the same type as x with CastLike

Reply:
I assume x always has dtype. Any case that x.dtype could be None?

Reply:
Not sure about the Nivida use case of this Pass but i added a check here.

Reply:
For model builder models there is always types. maybe an assert will do?

Reply:
Yeah they primarily want to handle model builder model.